### ISSUE TITLE
Project Secret Key grants the same permissions as the owner of the project.

### ISSUE THREAD

**Issue by devleejb** [yorkie-team/yorkie#813]
**What happened**:

The Project Secret Key grants the same permissions as the owner of the project. For example, when Admin A owns Project B and uses the Secret Key of that project, they can access other projects owned by Admin A as well. This occurs because accessing a project via the Project Secret Key returns the user who owns the project.

![image](https://github.com/yorkie-team/yorkie/assets/52884648/d78f073e-7bef-485e-a078-ce417db0bfdf)

**What you expected to happen**:

The Project Secret Key should only provide access to the specific project it belongs to and not grant access to other projects owned by the same user.

**How to reproduce it (as minimally and precisely as possible)**:

The issue can be reproduced by using the Project Secret Key to access a project and observing that it inadvertently provides access to other projects owned by the same user.

**Anything else we need to know?**:

It is suggested that a solution could involve adding a flag to the User to distinguish between Admin and Project access rights.

```go
claims, err := i.tokenManager.Verify(authorization)
if err == nil {
    user, err := users.GetUserByName(ctx, i.backend, claims.Username)
    if err == nil {
        return user, nil
    }
}

project, err := projects.GetProjectFromSecretKey(ctx, i.backend, authorization)
if err == nil {
    user, err := users.GetUserByID(ctx, i.backend, project.Owner)
    if err == nil {
        return user, nil // Need to return Project permissions, not Admin permissions
    }
}
``` 

**Environment**:
- Operating system: ALL
- Browser and version: ALL
- Yorkie version (use `yorkie version`): v0.4.14
- Yorkie JS SDK version: NONE

**Comment by hackerwins** [comment]
https://github.com/yorkie-team/yorkie/blob/e4885cba15cbac2d442b6b17217b43734d39cf4b/server/rpc/interceptors/admin.go#L192-L195

**Comment by dener8** [comment]
I've been assigned another issue, but can I try this first?

**Comment by devleejb** [comment]
@dener8 
Of course! If you want to unassign the Assignee of any other issue, just let me know.

**Comment by devleejb** [comment]
@dener8 
For CodePair, the component diagram is below.

![image](https://github.com/user-attachments/assets/6a29b5ba-7d8e-4ee2-8e62-c7965b72629a)

The comparison between the Secret Key and User Token defined in this issue is as follows.

| Item                    | As-Is                                  | To-Be                          |
|-------------------------|----------------------------------------|--------------------------------|
| **Project X's Secret Key** | Accessible to all projects of Developer A | Accessible to Project X only |
| **User Token of Developer A**          | Accessible to all projects of Developer A | Accessible to all projects of Developer A |

**Comment by window9u** [comment]
Hello  @dener8 😊
With the 2025 OSSCA project now active, we're looking to keep issues moving. If you're still actively working on this one, please let us know.
If not, we'll unassign you soon to open it up for others who are interested. Thanks for your understanding and contributions to Yorkie!

**Comment by SANGHEEJEONG** [comment]
I was assigned to #879, but it seems to be closely related to 813. Would it be okay to work on them together?

**Comment by hackerwins** [comment]
@SANGHEEJEONG Sure. If you have any questions, please feel free to ask.
https://discord.gg/MVEAwz9sBy
